src/dotnet/commands/dotnet-new/Program.cs

```diff
@@ -42,12 +37,34 @@ where resourceName.Contains(templateDir)
             {
                 string fileName = GetFileNameFromResourceName(resourceName);
 
-                resourceNameToFileName.Add(resourceName, fileName);
-                if (File.Exists(fileName))
+                using (var fileStream = File.Create(fileName))
```

Comment:
We don't need to create a physical file, just so we can unzip the template. System.IO.Compression has APIs that take in a stream, we should use those APIs.  That way we don't write a file, unzip it, and then delete the file.  Just unzip the stream directly.

See @ianhays if you need help.